File: C++/118.cpp
#include <iostream>
#include <vector>

using namespace std;

/*

*/
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res(numRows);
        for (int i = 0; i < numRows; i++) {
            res[i].resize(i+1, 1);
            for (int j = 1; j < i; j++) {   //跳过第一个和最后一个
                res[i][j] = res[i-1][j-1] + res[i-1][j];
            }     
        }
        return res;
    }
};

File: Python/118.py
from typing import List

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        res = []
        i = 1
        while numRows > 0:
            tmp = []
            if i == 1:
                tmp.append(1)
            elif i==2:
                tmp.append(1)
                tmp.append(1)
            else:
                tmp.append(1)
                last = res[-1]
                for j in range(len(last)-1):
                    tmp.append(last[j]+last[j+1])
                tmp.append(1)
            res.append(tmp)
            i += 1
            numRows -= 1
        return res

